src/edit_session/bracket_match.js

```diff
@@ -219,5 +219,190 @@ function BracketMatch() {
         
         return null;
     };
+
+    /**
+     * Returns [[Range]]'s for matching tags and tag names, if there are any
+     * @param {Position} pos
+     * @returns {{closeTag: Range, closeTagName: Range, openTag: Range, openTagName: Range} | undefined}
+     */
+    this.getMatchingTags = function (pos) {
+        var iterator = new TokenIterator(this, pos.row, pos.column);
+        var token = iterator.getCurrentToken();
+        var found = false;
+        var backward = false;
+        if (token && token.type.indexOf('tag-name') === -1) {
+            do {
+                if (backward) token = iterator.stepBackward(); else token = iterator.stepForward();
```

Comment:
is there a possibility that we enter an endless loop here? maybe we should mark visited tokens so we break in this case?

Reply:
Actually, I don't see possibility of endless loop here, because of loop end condition - checking availability of `token`.   This `backward` check set only for one case, when we are in self-closing tag in between tag name and tag closing. I could improve this by remembering initial token and changing token to it in case we changed direction (direction is changing only once) or just put some comment, explaining why we need this check.

Reply:
@mkslanc I guess having a comment can work as well so it's easier to understand the context here